tutorials/Tutorial5_Evaluation.py

```diff
@@ -2,63 +2,85 @@
     doc_index = "tutorial5_docs"
     label_index = "tutorial5_labels"
+
     ##############################################
     # Code
     ##############################################
+    launch_es()
     device, n_gpu = initialize_device_settings(use_cuda=True)
-    # Start an Elasticsearch server
-    # You can start Elasticsearch on your local machine instance using Docker. If Docker is not readily available in
-    # your environment (eg., in Colab notebooks), then you can manually download and execute Elasticsearch from source.
-    if LAUNCH_ELASTICSEARCH:
-        logging.info("Starting Elasticsearch ...")
-        status = subprocess.run(
-            ['docker run -d -p 9200:9200 -e "discovery.type=single-node" elasticsearch:7.9.2'], shell=True
-        )
-        if status.returncode:
-            raise Exception("Failed to launch Elasticsearch. If you want to connect to an existing Elasticsearch instance"
-                            "then set LAUNCH_ELASTICSEARCH in the script to False.")
-        time.sleep(30)
 
     # Download evaluation data, which is a subset of Natural Questions development set containing 50 documents
     doc_dir = "../data/nq"
     s3_url = "https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/nq_dev_subset_v2.json.zip"
     fetch_archive_from_http(url=s3_url, output_dir=doc_dir)
 
     # Connect to Elasticsearch
-    document_store = ElasticsearchDocumentStore(host="localhost", username="", password="", index="document",
-                                                create_index=False, embedding_field="emb",
-                                                embedding_dim=768, excluded_meta_data=["emb"])
-
+    document_store = ElasticsearchDocumentStore(
+        host="localhost", username="", password="", index="document",
+        create_index=False, embedding_field="emb",
+        embedding_dim=768, excluded_meta_data=["emb"]
+    )
 
     # Add evaluation data to Elasticsearch document store
     # We first delete the custom tutorial indices to not have duplicate elements
+    # and also split our documents into shorter passages using the PreProcessor
+    preprocessor = PreProcessor(
+        split_length=500,
+        split_overlap=0,
+        split_respect_sentence_boundary=False,
+        clean_empty_lines=False,
+        clean_whitespace=False
+    )
     document_store.delete_all_documents(index=doc_index)
     document_store.delete_all_documents(index=label_index)
-    document_store.add_eval_data(filename="../data/nq/nq_dev_subset_v2.json", doc_index=doc_index, label_index=label_index)
+    document_store.add_eval_data(
+        filename="../data/nq/nq_dev_subset_v2.json",
+        doc_index=doc_index,
+        label_index=label_index,
+        preprocessor=preprocessor
+    )
+
+    # Let's prepare the labels that we need for the retriever and the reader
+    labels = document_store.get_all_labels_aggregated(index=label_index)
+    q_to_l_dict = {
```

Comment:
I think we do not really add value here by splitting into retriever and reader labels since they refer to the same object anyways?
So lets keep it labels and add it as singular label in the run command.
If we add specific RetrieverLabels/ReaderLabels then we should pass them in the run method directly.

Reply:
As discussed, we will for now have a single labels dictionary that can contain various label objects. In future we will define more classes to structure these labels